### PR TITLE
Optimize query to prevent notification from mongo db

### PR DESCRIPTION
**Pull request by KarishmaBothara**
## Summary

Break look_up query and have two mongo api calls to get data from different collection instead of using join.

TRN-261

## Checklist

- [x] Add description

### CODE REVIEW

File: components/erc20TransfersForAddress.js [hunk 1]
```diff
@@ -32,6 +33,37 @@ export default function Erc20TransfersForAddress({ walletAddress }) {
 			refetchInterval: 15_000,
 		}
 	);
+	const queryContractDetails = useQuery(
+		["evm_transactions_contract_data", walletAddress],
+		() => {
+			return getContractDataForAddress(walletAddress);
+		},
+		{
+			refetchInterval: 85_000,
```
Comment (ken-centrality) [inline components/erc20TransfersForAddress.js:42]: the `refetchInterval` here is different to the one in `queryTxs`,  are we running into the risk of it's being out of sync?

File: components/evmTransactionsWidget.js [hunk 2]
```diff
@@ -19,6 +19,7 @@ export default function EvmTransactionsWidget() {
 			refetchInterval: 15_000,
 		}
 	);
+	console.log("Query....", query);
```
Comment (ken-centrality) [inline components/evmTransactionsWidget.js:22]:
```suggestion
```

File: pages/api/evm/getERC20TransferForAddress.ts [hunk 3]
```diff
@@ -65,30 +65,30 @@ export default async function handler(
...
-					},
-					{
-						$lookup: {
-							from: "Contractaddresses",
-							localField: "parsedLogs.args.to",
-							foreignField: "address",
-							as: "toContract",
-						},
-					},
+					// {
```
Comment (ken-centrality) [inline pages/api/evm/getERC20TransferForAddress.ts:68]: remove commented code

File: pages/api/evm/getERC721TransferForAddress.ts [hunk 4]
```diff
@@ -65,30 +65,30 @@ export default async function handler(
...
-					},
-					{
-						$lookup: {
-							from: "Contractaddresses",
-							localField: "toContract",
-							foreignField: "address",
-							as: "toContract",
-						},
-					},
+					// {
```
Comment (ken-centrality) [inline pages/api/evm/getERC721TransferForAddress.ts:68]: remove commented code

File: pages/api/evm/getEVMTransactions.ts [hunk 5]
```diff
@@ -33,22 +33,22 @@ export default async function handler(
...
-					},
-					{
-						$lookup: {
-							from: "Contractaddresses",
-							localField: "to",
-							foreignField: "address",
-							as: "toContract",
-						},
-					},
+					// {
```
Comment (ken-centrality) [inline pages/api/evm/getEVMTransactions.ts:36]: remove commented code

File: libs/evm-api/index.js [hunk 6]
```diff
@@ -63,3 +67,7 @@ export const getContractData = async (address) => {
 export const getEVMTransactions = async (page, limit) => {
 	return await fetchEvmData("getEVMTransactions", { page, limit });
 };
+
+export const getContractDataForAddresses = async (addresses) => {
```
Comment (ken-centrality) [inline libs/evm-api/index.js:71]: this is not used anywhere

File: pages/api/evm/getContractDataForAddresses.ts
Comment (ken-centrality) [inline pages/api/evm/getContractDataForAddresses.ts:1]: not used anywhere

**Review by ken-centrality — CHANGES_REQUESTED** [review]

File: components/erc20TransfersForAddress.js (hunk 1 above)
Comment (KarishmaBothara) [reply components/erc20TransfersForAddress.js:42]: This query gets the contract details.. if it exist so I made it refetch after 85_000 seconds.. cause most of the time data received from here would be null

File: libs/evm-api/index.js (hunk 6 above)
Comment (KarishmaBothara) [reply libs/evm-api/index.js:71]: will check this..

File: pages/api/evm/getContractDataForAddresses.ts
Comment (KarishmaBothara) [reply pages/api/evm/getContractDataForAddresses.ts:1]: same as above

**Review by ken-centrality — APPROVED** [review]

**Review by ken-centrality — APPROVED** [review]

**Pull request merged**